com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/monitoring/FlowableJobExecutorInformation.java

```diff
@@ -0,0 +1,49 @@
+package com.sap.cloud.lm.sl.cf.web.monitoring;
+
+import java.time.Instant;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.flowable.job.service.impl.asyncexecutor.DefaultAsyncJobExecutor;
+
+@Named
+public class FlowableJobExecutorInformation {
+
+    private static final int CACHE_TIMEOUT_IN_SECONDS = 60;
+
+    protected Instant lastUpdateTime;
+
+    private DefaultAsyncJobExecutor jobExecutor;
+
+    private int currentJobExecutorQueueSize = 0;
+
+    @Inject
+    public FlowableJobExecutorInformation(DefaultAsyncJobExecutor jobExecutor) {
+        this.jobExecutor = jobExecutor;
+    }
+
+    public int getCurrentJobExecutorQueueSize() {
+        if (lastUpdateTime == null || isPastCacheTimeout()) {
+            updateCurrentJobExecutorQueueSize();
+        }
+        return currentJobExecutorQueueSize;
+    }
+
+    private boolean isPastCacheTimeout() {
+        return Instant.now()
+                      .minusSeconds(CACHE_TIMEOUT_IN_SECONDS)
+                      .isAfter(lastUpdateTime);
+    }
+
+    private void updateCurrentJobExecutorQueueSize() {
+        currentJobExecutorQueueSize = jobExecutor.getThreadPoolQueue()
```

Comment:
Is this operation expensive enough to justify the cache?

Reply:
For every execution of size() method, it obtains lock over blocking queue and this could slow add/remove jobs from queue because of frequent polling of jmx beans from Dynatrace.